raytracing/matrixgroup.py

```diff
@@ -235,3 +235,119 @@ def __next__(self):
             self.iteration += 1
             return element
         raise StopIteration
+
+    def __len__(self):
+        return len(self.elements)
+
+    def __getitem__(self, item):
+        if isinstance(item, slice):  # If we get a slice, return a matrix group
+            return MatrixGroup(self.elements[item])
+        return self.elements[item]
+
+    def removeElement(self, index: int, pad: bool = False):
```

Comment:
We use `append()`, should we use `remove()` and `insert()` ? It is not coherent to have both I think.